src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -0,0 +1,533 @@
+        metric_id = _resolve(org_id, "user")
+        if metric_id is not None:
+            data.extend(get_query_data("metrics_sets", "user", metric_id, ["value"]))
+            metric_to_fields["user"] = [_UserField()]
+
+    duration_fields = [field for field in query.raw_fields if "session.duration" in field]
+    if duration_fields:
+        metric_id = _resolve(org_id, "session.duration")
+        if metric_id is not None:
+            columns = {"percentiles" if field[0] == "p" else field[:3] for field in duration_fields}
+
+            data.extend(
+                get_query_data(
+                    "metrics_distributions",
+                    "session.duration",
+                    metric_id,
+                    list(columns),
+                    # This metric does not have the status key
+                    remove_groupby={Column(f"tags[{tag_key_session_status}]")},
+                )
+            )
+            group_by_status = "session.status" in query.raw_groupby
+            metric_to_fields["session.duration"] = [
+                _SessionDurationField(field, group_by_status) for field in duration_fields
+            ]
+
+    if "sum(session)" in query.raw_fields:
+        metric_id = _resolve(org_id, "session")
+        if metric_id is not None:
+            if "session.status" in groupby:
+                # We need session counters grouped by status, as well as the number of errored sessions
+
+                # 1 session counters
+                data.extend(get_query_data("metrics_counters", "session", metric_id, ["value"]))
+
+                # 2: session.error
+                error_metric_id = _resolve(org_id, "session.error")
+                if error_metric_id is not None:
+                    groupby.pop("session.status")
+                    data.extend(
+                        get_query_data("metrics_sets", "session.error", error_metric_id, ["value"])
+                    )
+            else:
+                # Simply count the number of started sessions:
+                init = _resolve(org_id, "init")
+                if tag_key_session_status is not None and init is not None:
+                    extra_conditions = [
+                        Condition(Column(f"tags[{tag_key_session_status}]"), Op.EQ, init)
+                    ]
+                    data.extend(
+                        get_query_data(
+                            "metrics_counters", "session", metric_id, ["value"], extra_conditions
+                        )
+                    )
+
+            metric_to_fields["session"] = [_SumSessionField()]
+
+    return data, metric_to_fields
+
+
+def _flatten_data(org_id: int, data: _SnubaDataByMetric) -> _FlatData:
+    """Unite snuba data from multiple queries into a single key-value map for easier access"""
+    flat_data = {}
+
+    # It greatly simplifies code if we just assume that these two tags exist:
+    # TODO: Can we get away with that assumption?
```

Comment:
let's punt on this. we have to answer the same question everywhere else. I've created a JIRA task for it